operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -268,8 +268,11 @@ private String getLockName(String namespace, String name) {
 
         Promise<Void> result = Promise.promise();
         handler.onComplete(reconcileResult -> {
-            handleResult(reconciliation, reconcileResult, reconciliationTimerSample);
-            result.handle(reconcileResult);
+            try {
+                handleResult(reconciliation, reconcileResult, reconciliationTimerSample);
+            } finally {
+                result.handle(reconcileResult);
```

Comment:
did you think about whether it would it be more correct to fail the promise in the case of a throwable?

Reply:
I considered that, but thought the `handleResult` processing is more bookkeeping than reconciliation. It also seems that the result of the promise is generally ignored outside of test code.